server/controllers/shoppingList.js

```diff
@@ -149,8 +181,10 @@ const getProductsForGivenList = (req, resp) => {
         return resp.status(404).send({ message: err.message });
       }
 
-      const { products } = documents[0];
-      resp.status(200).json(products);
+      if (documents[0]) {
```

Comment:
do you need check it aslo here? Didn't first `if` checked if something was found?